tools/wake-format/emitter.cpp

```diff
@@ -918,9 +930,12 @@ wcl::doc Emitter::walk_block(ctx_t ctx, CSTElement node) {
 
   // clang-format off
   auto body_fmt = fmt().match(
-    // TODO: starting 'pred()' function doesn't allow init lists
-    pred(ConstPredicate(false), fmt())
-   .pred({TOKEN_WS, TOKEN_NL, TOKEN_COMMENT}, fmt().next())
+    // Block level newlines should be re-emitted as these are the
+    // user creating 'pseudo-blocks'
+    pred(TOKEN_NL, fmt().newline().newline().next())
+   .pred(TOKEN_WS, fmt().next())
+   // Comments are following by a block level newline that shouldn't be emitted
+   .pred(TOKEN_COMMENT, fmt().next().next())
```

Comment:
Is this meant to be "followed by"?  Either way, it looks like none of the in-block comments being tested have an author-provided newline next to them (neither before nor after), so that might be something we want to add to the test input.

Reply:
>Is this meant to be "followed by"?

yeah, was a typo, nice catch.

>  Either way, it looks like none of the in-block comments being tested have an author-provided newline next to them 

That isn't the case being covered here. Every comment must be followed by a single newline to end the comment

```
def a = b # comment <nl>
def c = d # comment <nl>
```
since we always emit a newline when emitting a comment, we need to skip it here so that it isn't duplicated. In the case of a author added newline we'd have

```
def a = b # comment <nl>
<nl>
def c = d # comment <nl>
```

which would get covered by the line#935 predicate

> so that might be something we want to add to the test input.

yeah, that's a good idea. I'll add it

Reply:
Ah, yeah, that makes sense.  I think I was getting thrown off by the "block level newline" terminology.  And with the new test (and associated fix it seems to have revealed), this all looks good!